reference/basic-architecture.md

```diff
@@ -9,61 +9,62 @@ More detailed guides to Mayastor's components, their design and internal structu
-
-It is not necessary for the node plugin to run on every worker node within a cluster and this behaviour can be modified if so desired through the application of appropriate node labeling and the addition of a corresponding `nodeSelector` entry within the pod spec of the mayastor-csi DaemonSet. It should be noted that if a node does not host a plugin pod, then it will not be possible to schedule pod on it which is configured to mount Mayastor volumes.
-
-Further detail regarding the implementation of CSI driver components and their function can be found within the Kubernetes CSI Developer Documentation.
-
-### NATS
+### etcd
+   
+etcd is a distributed reliable key-value store for the critical data of a distributed system. Mayastor uses etcd as a reliable persistent store for its configuration and state data.
 
-[NATS](https://nats.io/) is a high performance open source messaging system. It is used by Mayastor as the transport mechanism for registration messages passed between Mayastor I/O engine pods running in the cluster and the core agent component, which maintains an inventory of active Mayastor nodes.  It is also used as the transport abstraction for inter data-plane / control-plane gRPC calls.
+### Supportability:
 
-### etcd
+The supportability tool is used to create support bundles (archive files) by interacting with multiple services present in the system. These bundles contain information about the entire Mayastor system, as well as specific Mayastor resources like volumes, pools and nodes, and can be used for debugging. It can collect the following information:
+- Topological information of Mayastor's resource(s) by interacting with the REST service
+- Historical logs by interacting with Loki. If Loki is unavailable, it interacts with the kube-apiserver to fetch logs.
+- Mayastor-specific Kubernetes resources by interacting with the kube-apiserver
+- Mayastor-specific information from etcd (internal) by interacting with the etcd server.
 
-"[etcd](https://github.com/etcd-io/etcd) is a distributed reliable key-value store for the most critical data of a distributed system"
+### Loki:
 
-Mayastor uses etcd as a reliable persistent store for its configuration and state data.
+    [Loki](https://grafana.com/oss/loki/) aggregates and centrally stores logs from all Mayastor containers which are deployed in the cluster.
 
+### Promtail:
+     
+    Promtail is a log collector built specifically for Loki. It uses the configuration file for target discovery and includes analogous features for labeling, transforming, and filtering logs from containers before ingesting them to Loki.
```

Comment:
For the sake of heirarchy, I think we should keep Loki and Promtail, and also kubectl supportability commands under the supportability section giving the notion of a supportability stack.

Reply:
Based on our call earlier, I will be addressing these as part of next PR.